### PR TITLE
npm: Updated http-api and api packages

### PR DESCRIPTION
**Pull request by tylershuster**
This brings the `@urbit/api` and `@urbit/http-api` code in the code base up to date with what is currently published on npm. 

Landscape does not rely on `http-api` so those changes shouldn't be a problem — most of these changes are bugifxes discovered while working on https://github.com/urbit/urbit/pull/4611 or by other contributors.

Landscape does rely on `api` but most of those changes should be to functions that have not been integrated back into Landscape yet.

### CODE REVIEW

**Comment by tylershuster** [comment]
Generally, I want to unblock other work that relies on this to make the eventual migration to `http-api` easier.

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
One type error, looks good otherwise

File: pkg/npm/api/metadata/types.ts [hunk 1]
```diff
@@ -67,8 +65,9 @@ export interface Metadata {
   'date-created': string;
   description: string;
   title: string;
-  module: string;
+  config: { graph: string };
```
[the remark this answers is not in the document]
Comment (tylershuster) [reply pkg/npm/api/metadata/types.ts:68]: as in:

```ts
config: {
  graph?: string;
  group?: {} | Resource;
}
```

[the remark this answers is not in the document]
Comment (tylershuster) [reply pkg/npm/api/metadata/types.ts:68]: @liam-fitzgerald ?

[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply pkg/npm/api/metadata/types.ts:68]: no:
```typescript
interface Metadata {
// elided
  config: MetadataConfig;
}

type MetadataConfig = GroupConfig | GraphConfig;
interface GroupConfig {
  group: null | {} | Resource;
}
interface GraphConfig {
  graph: string;
}
```
```

**Comment by tylershuster** [comment]
Ok, then this looks g2g

**Review by matildepark — APPROVED** [review]

**Review by liam-fitzgerald — APPROVED** [review]

**Pull request merged**